cfnstack/provisioner.go

```diff
@@ -231,3 +231,67 @@ func (c *Destroyer) Destroy() error {
 	_, err := cfSvc.DeleteStack(dreq)
 	return err
 }
+
+func (c *Provisioner) StreamEventsNested(q chan struct{}, f *cloudformation.CloudFormation, stackId string, headStackName string, t time.Time) error {
+	nestedStacks := make(map[string]bool)
+	nestedQuit := make(chan struct{}, 1)
+	var lastSeenEventId string
+	defer func() { nestedQuit <- struct{}{} }()
+	for {
+		select {
+		case <-q:
+			return nil
+		case <-time.After(1 * time.Second):
+			events := make([]cloudformation.StackEvent, 0)
+
+			_ = f.DescribeStackEventsPages(
+				&cloudformation.DescribeStackEventsInput{StackName: &stackId},
+				func(page *cloudformation.DescribeStackEventsOutput, lastPage bool) bool {
+					for _, e := range page.StackEvents {
+						if (*e.Timestamp).Before(t) {
```

Comment:
I wonder if pointer dereference is actually needed here and throughout the function? Doesn't Go dereference them on access and implicit dereferencing is needed only on assignments?